index.js

```diff
@@ -198,8 +198,9 @@ function alloc (size, fill, encoding) {
     // prevents accidentally sending in a number that would
     // be interpretted as a start offset.
     return typeof encoding === 'string'
-      ? createBuffer(size).fill(fill, encoding)
-      : createBuffer(size).fill(fill)
+      // cast boolean to unsigned
+      ? createBuffer(size).fill(typeof fill === 'boolean' ? Number(fill) : fill, encoding)
+      : createBuffer(size).fill(typeof fill === 'boolean' ? Number(fill) : fill)
```

Comment:
Instead of repeating the `typeof fill === 'boolean'` twice, can you pull this out into a line above the return statement?

```js
if (typeof fill === 'boolean') fill = Number(fill)
```